packages/eslint-plugin/src/rules/dot-notation.ts

```diff
@@ -89,7 +90,17 @@ export default createRule<Options, MessageIds>({
           node.computed
         ) {
           // for perf reasons - only fetch symbols if we have to
-          const propertySymbol = services.getSymbolAtLocation(node.property);
+          const propertySymbol =
+            services.getSymbolAtLocation(node.property) ??
```

Comment:
Unit tests still pass without this line. Either there's a missing unit test or this line can be removed.

```suggestion
```